Add CLI contao:user command (#1413)
contao/docs: docs/manual/cli/user.en.md

---
title: "contao:user"
description: "Create and list Contao users."
aliases:
    - /en/cli/user/
---

After the [database](/en/cli/migrate/) has been created, Contao [users](/en/user-management/users/) can be created here. You can get 
an overview of the commands via:

```bash
php vendor/bin/contao-console contao:user --help

> ```

Reviewer: Such a command does not actually exist - i.e. there is no actual `contao:user` command.

Author: But lists me the available commands for the "contao:user" namespace.

Author: > For the contao:user:password command I would also mention the required username argument.

Added

Author: > For the contao:user:create command I would also mention, that it will interactively ask you for all the user's details when executed without any options.

Added

Reviewer: > But lists me the available commands for the "contao:user" namespace

It shows an error that the command does not exist.

Author: Deleted.

## contao:user:list

Lists existing Contao users:

> | Option | Beschreibung |

Reviewer 2: ```suggestion
| Option | Description |
```

| --- | --- |
| `--help`   | Options overview |

> | `--admins` | Show administrators only |

Reviewer 3: ```suggestion
| `--column=COLUMN` | The columns display in the table (multiple values are allowed) |
| `--admins` | Show administrators only |
| `--format=FORMAT` | Output format (txt, json) |
```

There is two more options that could be added.
I guess this example could kinda explain it:
```cli
contao:user:list --column=name --column=admin --format=json
```
returns
```json
[{"name":"Contao","admin":1},{"name":"Demo","admin":0}]
```

```bash
php vendor/bin/contao-console contao:user:list --help
```

## contao:user:create

Creates a Contao user:

> | Option | Beschreibung |

Reviewer 2: ```suggestion
| Option | Description |
```

| --- | --- |
| `--help`   | Options overview |
| `--username` | User name |
| `--name` | Full user name |
| `--email` | The user e-mail address |
| `--password` | User password |
| `--admin` | User is administrator |

> | `--group` | Groups to be assigned to the user |

Reviewer 3: Would be good to explain that you could use:
```contao:user:create --group=1 --group=3 --group=5```
to assign multiple groups at once as well

| `--change-password` | Prompt users to change their password when they first log in to the backend |

```bash
php vendor/bin/contao-console contao:user:create --help
```

## contao:user:password

> Change of a user password:
> 
> ```bash
> php vendor/bin/contao-console contao:user:password --help
> ```

Reviewer 3: ```suggestion
Allows modifiying the password for a user

| Option | Description |
| --- | --- |
| `--help`   | Options overview |
| `--username` | The username of the back end user |
| `--password`** | The new password (using this option is not recommended for security reasons) |
| `--p` | The new password (using this option is not recommended for security reasons) |
| `--require-change` | Require the user to change the password on their next login. |
| `--r` | Require the user to change the password on their next login. |

Example usage:
``
php vendor/bin/contao-console contao:user:password  -p mysecurepassword -r k.jones
``
```

Reviewer: For security reasons you should not pass the password as an option directly on the CLI, since the password will then be stored in plain text in the bash history.

Reviewer 3: Yup, better leave it undocumented then.
That's why I added the `(using this option is not recommended for security reasons)`

Reviewer: The option itself is fine, you should just not use it directly on the CLI.

---
title: "contao:user"
description: "Create and list Contao users."
aliases:
    - /en/cli/user/
---

After the [database](/en/cli/migrate/) has been created, Contao [users](/en/user-management/users/) can be created here.

## contao:user:list

Lists existing Contao users:

| Option | Description |
| --- | --- |
| `--help`   | Options overview |
| `--admins` | Show administrators only |

> | `--column=COLUMN` | The columns display in the table ( multiple values are allowed ) |

Reviewer 4: ```suggestion
| `--column=COLUMN` | The columns display in the table (multiple values are allowed) |
```

| `--format=FORMAT` | Output format ( txt, json ) |

```bash
php vendor/bin/contao-console contao:user:list --help
```

## contao:user:create

Creates a Contao user. If the command is executed without options, the user is interactively prompted for all details.

| Option | Description |
| --- | --- |
| `--help`   | Options overview |
| `--username` | User name |
| `--name` | Full user name |
| `--email` | The user e-mail address |
| `--password` | User password |
| `--admin` | User is administrator |

> | `--group` | Groups to be assigned to the user ( multiple values are allowed ) |

Reviewer 4: ```suggestion
| `--group` | Groups to be assigned to the user (multiple values are allowed) |
```

| `--change-password` | Prompt users to change their password when they first log in to the backend |

```bash
php vendor/bin/contao-console contao:user:create --help
```

## contao:user:password

> Change of a user password ( requires a username as an argument ). If the command is executed without options, 

Reviewer 4: ```suggestion
Change of a user password (requires a username as an argument). If the command is executed without options, 
```

the required information is requested interactively.

```bash
php vendor/bin/contao-console contao:user:password --help
```

| Option | Description |
| --- | --- |
| `--help`   | Options overview |
| `--username` | The username of the back end user |

> | `--password` | The new password ( using this option is not recommended for security reasons ) |

Reviewer 4: ```suggestion
| `--password` | The new password (using this option is not recommended for security reasons) |
```
